src/frontend/fluent-core/App.ts

```diff
@@ -0,0 +1,144 @@
+    }
+
+    #titlebar {
+    }
+
+    #panel {
+        display: flex;
+        flex-direction: row;
+        gap: calc(var(--base-height-multiplier) * 1px);
+        margin: calc(var(--base-height-multiplier) * 1px);
+    }
+
+    #sidepanel {
+        flex: 2;
+        display: flex;
+        flex-direction: column;
+        gap: calc(var(--base-height-multiplier) * 1px);
+    }
+
+    #mainpanel {
+        flex: 3;
+        gap: calc(var(--base-height-multiplier) * 1px);
+        display: grid;
+        grid-template-columns: auto;
+        grid-template-rows: min-content min-content minmax(0, 1fr);
+    }
+
+    #bookmark-list-panel {
+        flex: 1;
+        min-height: 0;
+        /*border: 1px dotted green;*/
+    }
+
+    #download-manager-panel {
+        flex: 1;
+        min-height: 0;
+        /*border: 1px dotted blue;*/
+    }
+
+    #bookmark-list, #download-manager, #website-select, #media-title-select, #media-item-list {
+        height: 100%;
+    }
+`;
+
+const templateSidePanel: ViewTemplate<App> = html`
+    <div id="sidepanel">
+        <fluent-card id="bookmark-list-panel" style="display: ${() => S.SettingPanelBookmarks ? 'block' : 'none'}">
+            <fluent-bookmark-list id="bookmark-list" @bookmarkClicked=${(model, ctx) => model.BookmarkClicked(ctx.event)}></fluent-bookmark-list>
+        </fluent-card>
+        <fluent-card id="download-manager-panel" style="display: ${() => S.SettingPanelDownloads ? 'block' : 'none'}">
+            <fluent-download-manager id="download-manager"></fluent-download-manager>
+        </fluent-card>
+    </div>
+`;
+
+const template: ViewTemplate<App> = html`
+    <fluent-titlebar id="titlebar"></fluent-titlebar>
+    <div id="panel">
+        ${when(() => S.SettingPanelBookmarks || S.SettingPanelDownloads, templateSidePanel)}
+        <div id="mainpanel">
+            <fluent-card>
+                <fluent-website-select id="website-select" :selected=${model => model.website}
+                    @selectedChanged=${(model, ctx) => model.WebsiteSelectedChanged(ctx.event)}
+                    @entriesUpdated=${(model, ctx) => model.WebsiteEntriesUpdated(ctx.event)}>
+                </fluent-website-select>
+            </fluent-card>
+            <fluent-card>
+                <fluent-media-title-select id="media-title-select" :entries=${model => model.titles}
+                    @selectedChanged=${(model, ctx) => model.MediaTitleSelectedChanged(ctx.event)}
+                    @entriesUpdated=${(model, ctx) => model.MediaTitleEntriesUpdated(ctx.event)}>
+                </fluent-media-title-select>
+            </fluent-card>
+            <fluent-card>
+                <fluent-media-item-list id="media-item-list" :entries=${model => model.items}></fluent-media-item-list>
+            </fluent-card>
+        </div>
+    </div>
+`;
+
+@customElement({ name: 'fluent-app', template, styles })
+export default class App extends FASTElement {
+
+    @observable website: IMediaContainer;
```

Comment:
May consider moving the shared observables to [StateService.ts](https://github.com/manga-download/haruneko/pull/53/files#diff-488cba7ae975394ca97a92f0e8e5105454897a8fcac476d378bc51302ae6ac17)